cgfs.py

```diff
@@ -503,26 +653,46 @@ def unlink(self, path):
         fname = parts[-1]
         file = self.get_file(fname, start=parent, expect_type=File)
 
-        try:
-            cgapi.delete_file(file.id)
-        except CGAPIException as e:
-            handle_cgapi_exception(e)
+        if isinstance(file, TempFile):
+            file.unlink()
+        else:
+            if self.fixed:
+                raise FuseOSError(EPERM)
+
+            try:
+                cgapi.delete_file(file.id)
+            except CGAPIException as e:
+                handle_cgapi_exception(e)
 
         parent.pop(fname)
 
     def utimens(self, path, times=None):
         file = self.get_file(path)
 
+        assert file is not None
+
         atime, mtime = times or (time(), time())
-        file.setattr('st_atime', atime)
-        file.setattr('st_mtime', mtime)
+
+        if isinstance(file, TempFile):
+            file.utimens(atime, mtime)
+        elif self.fixed:
+            raise FuseOSError(EPERM)
+        else:
+            file.setattr('st_atime', atime)
+            file.setattr('st_mtime', mtime)
```

Comment:
Shouldn't we add a `utimens` method on the `File` class instead of having a special case for `TempFile`?